test/config/karma.conf.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+var webpackConfig = require('../../webpack.test');
+
+module.exports = function(config) {
+  var singleRun = process.env.SINGLE_RUN ? process.env.SINGLE_RUN !== 'false' : true;
+
+  config.set({
+    basePath: '../../',
+    files: [
+      'src/index.test.js',
+      'test/fixtures/*.js',
+      'test/unit-frontend/fixtures/**',
+      'src/linagora.esn.calendar/app/fixtures/**',
+      '**/*.pug'
+    ],
+    exclude: [
+      'node_modules/esn-frontend-common-libs/src/frontend/**/*.spec.js',
+      'node_modules/esn-frontend-common-libs/src/frontend/js/esn/app.js',
+      'src/linagora.esn.calendar/app/search/search.run.js',
+      'src/linagora.esn.calendar/app/module-registry.run.js',
+      'src/linagora.esn.calendar/app/configuration/configuration.run.js',
+      'src/linagora.esn.calendar/app/services/cal-default-value.run.js',
+      'src/linagora.esn.calendar/app/services/websocket/listener.run.js',
+      'src/linagora.esn.calendar/event-consultation-app/app.js',
+      'src/linagora.esn.calendar/event-consultation-app/error.js',
+      'src/linagora.esn.calendar/event-consultation-app/error-display/**/*.js'
+    ],
+    frameworks: ['mocha', 'sinon-chai'],
+    colors: true,
+    singleRun: singleRun,
+    autoWatch: true,
+    browsers: ['FirefoxHeadless'/*,'ChromeHeadless'*/],
+
+    customLaunchers: {
+      FirefoxHeadless: {base: 'Firefox', flags: ['--headless']},
+      ChromeHeadless: {base: 'Chrome', flags: ['--headless', '--disable-gpu']},
+      Chrome_with_debugging: {
+        base: 'Chrome',
+        flags: ['--remote-debugging-port=9222'],
+        debug: true
+      }
+    },
+
+    proxies: {
+      '/images/': '/base/frontend/images/'
+    },
+
+    reporters: singleRun ? ['coverage', 'spec'] : ['spec'],
+
+    preprocessors: {
+      'src/index.test.js': ['webpack'],
+      '**/*.pug': ['ng-jade2module'],
+      'test/fixtures/**': ['raw2js'],
+      'src/linagora.esn.calendar/app/fixtures/**': ['raw2js'],
+      'test/unit-frontend/fixtures/**': ['raw2js']
+    },
+
+    webpack: webpackConfig,
+    plugins: [
+      'karma-chrome-launcher',
+      'karma-firefox-launcher',
+      'karma-mocha',
+      'karma-webpack',
+      'karma-coverage',
+      'karma-spec-reporter',
+      'karma-sinon-chai',
+      '@linagora/karma-ng-jade2module-preprocessor',
+      'karma-rawfixtures-preprocessor'
+    ],
+
+    coverageReporter: {type: 'text', dir: '/tmp'},
+
+    ngJade2ModulePreprocessor: {
```

Comment:
This is not needed because of webpack.

Reply:
We can only remove the `cacheIdfromPath` function. Others are still in need because we need sth to render our pug file. Try remove them and you will catch some errors i think.